lib/sendgrid_actionmailer.rb

```diff
@@ -197,7 +185,7 @@ def add_personalizations(sendgrid_mail, mail)
       end
 
       if mail['personalizations']
-        personalizations = json_parse('[' + mail['personalizations'].value + ']', false)
+        personalizations = mail['personalizations'].respond_to?(:unparsed_value) ? mail['personalizations'].unparsed_value.to_a : mail['personalizations'].map(&:unparsed_value)
```

Comment:
It's not entirely clear to me what's going on here; what is `mail["personalizations"]` expected to contain?

Would something like this do the same thing:

    personalizations = Array(mail["personalizations"]).map(&:unparsed_value)

?

Reply:
This was due to the fact that mail['personalizations'] could have been an array of mail inputs, or a single mail input, but restructuring add_personalization to apply personalizations before mail attributes resolves this issue and cleans up the code further.